nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c

```diff
@@ -509,206 +332,6 @@ doOpenFont(ClientPtr client, OFclosurePtr c)
-    ** satisfy the request with it.  Instead, we pass the FontPtr to the
-    ** FPE's open_font code (the fontfile FPE in turn passes the
-    ** information to the rasterizer; the fserve FPE ignores it).
-    **
-    ** Presumably, the font is marked non-cachable because the FPE has
-    ** put some licensing restrictions on it.  If the FPE, using
-    ** whatever logic it relies on, determines that it is willing to
-    ** share this existing font with the client, then it has the option
-    ** to return the FontPtr we passed it as the newly-opened font.
-    ** This allows the FPE to exercise its licensing logic without
-    ** having to create another instance of a font that already exists.
-    */
-
-	cached = FindCachedFontPattern(patternCache, pfontname, lenfname);
-	if (cached && cached->info.cachable)
-	{
-	    if (!AddResource(fid, RT_FONT, (void *) cached))
-		return BadAlloc;
-	    cached->refcnt++;
-	    return Success;
-	}
-    }
-    c = (OFclosurePtr) xalloc(sizeof(OFclosureRec));
-    if (!c)
-	return BadAlloc;
-    c->fontname = (char *) xalloc(lenfname);
-    c->origFontName = pfontname;
-    c->origFontNameLen = lenfname;
-    if (!c->fontname) {
-	xfree(c);
-	return BadAlloc;
-    }
-    /*
-     * copy the current FPE list, so that if it gets changed by another client
-     * while we're blocking, the request still appears atomic
-     */
-    c->fpe_list = (FontPathElementPtr *)
-	xalloc(sizeof(FontPathElementPtr) * num_fpes);
-    if (!c->fpe_list) {
-	xfree(c->fontname);
-	xfree(c);
-	return BadAlloc;
-    }
-    memmove(c->fontname, pfontname, lenfname);
-    for (i = 0; i < num_fpes; i++) {
-	c->fpe_list[i] = font_path_elements[i];
-	UseFPE(c->fpe_list[i]);
-    }
-    c->client = client;
-    c->fontid = fid;
-    c->current_fpe = 0;
-    c->num_fpes = num_fpes;
-    c->fnamelen = lenfname;
-    c->slept = FALSE;
-    c->flags = flags;
-    c->non_cachable_font = cached;
-
-    (void) doOpenFont(client, c);
-    return Success;
-}
-
-/**
- * Decrement font's ref count, and free storage if ref count equals zero
- *
- *  \param value must conform to DeleteType
- */
-int
-CloseFont(void * value, XID fid)
-{
-    int         nscr;
-    ScreenPtr   pscr;
-    FontPathElementPtr fpe;
-    FontPtr     pfont = (FontPtr)value;
-
-    if (pfont == NullFont)
-	return (Success);
-    if (--pfont->refcnt == 0) {
-	if (patternCache)
-	    RemoveCachedFontPattern (patternCache, pfont);
-	/*
-	 * since the last reference is gone, ask each screen to free any
-	 * storage it may have allocated locally for it.
-	 */
-	for (nscr = 0; nscr < screenInfo.numScreens; nscr++) {
-	    pscr = screenInfo.screens[nscr];
-	    if (pscr->UnrealizeFont)
-		(*pscr->UnrealizeFont) (pscr, pfont);
-	}
-	if (pfont == defaultFont)
-	    defaultFont = NULL;
-#ifdef LBX
-	LbxFreeFontTag(pfont);
-#endif
-#ifdef XF86BIGFONT
-        {
-           extern void XF86BigfontFreeFontShm(FontPtr);
```

Comment:
```
-     extern void XF86BigfontFreeFontShm(FontPtr);
```

> Changed here, is leaving this out okay?

The compiler should warn with implicit declaration, which it does not do. So I guess it is ok.